lock-api/src/main/java/com/palantir/lock/client/MultiClientTimeLockUnlocker.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.client;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+import com.palantir.atlasdb.autobatch.Autobatchers;
+import com.palantir.atlasdb.autobatch.BatchElement;
+import com.palantir.atlasdb.autobatch.DisruptorAutobatcher;
+import com.palantir.atlasdb.futures.AtlasFutures;
+import com.palantir.atlasdb.timelock.api.ConjureLockTokenV2;
+import com.palantir.atlasdb.timelock.api.ConjureUnlockRequestV2;
+import com.palantir.atlasdb.timelock.api.ConjureUnlockResponseV2;
+import com.palantir.atlasdb.timelock.api.Namespace;
+import com.palantir.common.streams.KeyedStream;
+import com.palantir.lock.v2.LockToken;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.OptionalInt;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import org.immutables.value.Value;
+
+public class MultiClientTimeLockUnlocker implements AutoCloseable {
+    private final DisruptorAutobatcher<NamespacedLockSet, Set<LockToken>> batcher;
+
+    public MultiClientTimeLockUnlocker(InternalMultiClientConjureTimelockService delegate, OptionalInt bufferSize) {
+        this.batcher = Autobatchers.independent(new UnlockConsumer(delegate))
+                .bufferSize(bufferSize)
+                .batchFunctionTimeout(Duration.ofSeconds(30))
```

Comment:
I think the unlock requests can be chonkier than other endpoints we have batched across namespaces. If the bufferSize chosen after taking into request/response size into account?

Reply:
Yes, that's a valid flag. Though that's probably something on the AtlasDB Proxy side to decide: a lock token is a UUID and so 36 bytes in string form; let's round that up to 40 to account for json overheads. So if we have a 50 MB limit on requests, we can get away with 1.25MM lock tokens. While it's hard to bound the size of an individual request, 16,384 would mean a given unlock request can have about 76 tokens; we can choose a lower constraint for this batcher and I'd think even say 1,024 for the ring buffer size would already give us some improvements.

Reply:
Thanks! That makes sense. As long as we do not fall back on default which I _think_ is 4096 in internal proxy (which also seems like a reasonable estimate)